From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Fri, 24 Jul 2015 13:02:59 +0200
Subject: [PATCH] backlight: pwm: add debug information for initial state
 decision

Add debug information showing how the decision to initially enable or
disable the backlight was made.

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
---
 drivers/video/backlight/pwm_bl.c | 21 ++++++++++++++++++---
 1 file changed, 18 insertions(+), 3 deletions(-)

diff --git a/drivers/video/backlight/pwm_bl.c b/drivers/video/backlight/pwm_bl.c
index 65e57da206e3..f4ca3d1560d6 100644
--- a/drivers/video/backlight/pwm_bl.c
+++ b/drivers/video/backlight/pwm_bl.c
@@ -266,16 +266,25 @@ static int pwm_backlight_probe(struct platform_device *pdev)
 		pb->enable_gpio = gpio_to_desc(data->enable_gpio);
 	}
 
+	dev_dbg(&pdev->dev, "%scontrolled via phandle\n", phandle ? "" : "not ");
+
 	if (pb->enable_gpio) {
+		bool output, enabled;
+
+		output = gpiod_get_direction(pb->enable_gpio) == GPIOF_DIR_OUT;
+		enabled = gpiod_get_value(pb->enable_gpio);
+		dev_dbg(&pdev->dev, "enable GPIO direction is %sput\n", output ?
+			"out" : "in");
+		dev_dbg(&pdev->dev, "enable GPIO value is %sabled\n", enabled ?
+			"en" : "dis");
+
 		/*
 		 * If the driver is probed from the device tree and there is a
 		 * phandle link pointing to the backlight node, it is safe to
 		 * assume that another driver will enable the backlight at the
 		 * appropriate time. Therefore, if it is disabled, keep it so.
 		 */
-		if (phandle &&
-		    gpiod_get_direction(pb->enable_gpio) == GPIOF_DIR_OUT &&
-		    gpiod_get_value(pb->enable_gpio) == 0)
+		if (phandle && output && !enabled)
 			initial_blank = FB_BLANK_POWERDOWN;
 		else
 			gpiod_direction_output(pb->enable_gpio, 1);
@@ -287,9 +296,15 @@ static int pwm_backlight_probe(struct platform_device *pdev)
 		goto err_alloc;
 	}
 
+	dev_dbg(&pdev->dev, "power supply regulator is %sabled\n",
+		regulator_is_enabled(pb->power_supply) ? "en" : "dis");
+
 	if (phandle && !regulator_is_enabled(pb->power_supply))
 		initial_blank = FB_BLANK_POWERDOWN;
 
+	dev_dbg(&pdev->dev, "backlight will be initially %sabled\n",
+		(initial_blank == FB_BLANK_UNBLANK) ? "en" : "dis");
+
 	pb->pwm = devm_pwm_get(&pdev->dev, NULL);
 	if (IS_ERR(pb->pwm)) {
 		ret = PTR_ERR(pb->pwm);
